tools/single_use/Engine_Change_MyISAM_to_INNODB.php

```diff
@@ -0,0 +1,212 @@
+ database with a MyISAM engine to use INNODB.
+This script includes foreign keycheck disabling and re-enabling.
+#####################################################################################
+\n\n";
+
+// define the command line parameters
+// expected commands all have 2 parameters
+if (count($argv) != 3 || $argv[1] == "help") {
+    showHelp();
+}
+
+// set default arguments
+$confirm    = false;
+$printToSQL = false;
+$output     = "";
+$action     = "";
+// array to hold all tables that will be converted
+$tablesToChange = array();
+
+// SETUP INPUT from Database table schema. used for `allTables` and `tableName` options
+$table_names = $DB->pselectCol(
+    "SELECT TABLE_NAME 
+     FROM INFORMATION_SCHEMA.TABLES
+     WHERE TABLE_SCHEMA =:dbn 
+       AND ENGINE = 'MyISAM'",
+    array("dbn" => $dbConfig["database"])
+);
+
+// possible actions for script are schemaTables, all Tables or any table in the DB
+$actions    = array("allTables","schemaTables") + $table_names;
+// END INPUT
+
+
+// get command arguments and validate
+if (in_array($argv[1],$actions,true)) {
+    $action = $argv[1];
+} else {
+    echo "ERROR: The action you have selected is invalid. Use `schemaTable, `allTables `
+    or specify the name of the table you would like to convert.\n\n\n";
+    showHelp();
+}
+
+if ($argv[2] === "confirm") {
+    $confirm = true;
+} elseif ($argv[2] === "tosql") {
+    $printToSQL = true;
+} else {
+    showHelp();
+}
+
+// If action is `schemaTables` only, parse schema files
+if ($action === "schemaTables") {
+    // SETUP INPUT from schema files in SQL loris directory
+    $schemaFiles = [
+        "0000-00-00-schema.sql",
+        "0000-00-01-Permission.sql",
+        "0000-00-02-Menus.sql",
+        "0000-00-03-ConfigTables.sql",
+        "0000-00-04-Help.sql",
+    ];
+    $schemaFileBase = $base . "SQL/";
+    $completeSchema = "";
+    foreach ($schemaFiles as $file) {
+        $completeSchema .= $fc = file_get_contents($schemaFileBase . $file, "r");
+    }
+    $createTables = [];
+    preg_match_all('/CREATE TABLE \`([_a-zA-Z0-9]+)\`/', $completeSchema, $createTables);
+
+    // Format array to match expectations
+    foreach ($createTables[1] as $key=>$table)
+    {
+        $tablesToChange[] = $table;
+    }
+    // END INPUT
+} elseif ($action === "allTables") {
+    $tablesToChange = $table_names;
+} else {
+    // its a specific table
+    $tablesToChange[] = $action;
+}
+
+if (empty($tablesToChange)) {
+    echo "No MyISAM tables detected. Terminating the script.\n";
+    die();
+}
+
+// SETUP OUTPUT
+$output .="SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS; \n";
+$output .="SET FOREIGN_KEY_CHECKS=0; \n";
+
+foreach ($tablesToChange as $table)
+{
+    $output .= "ALTER TABLE `".$table."` ENGINE=INNODB;\n";
+}
+
+$output .="SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS; \n";
+// END OUTPUT
+
+// START ACTION
+if ($printToSQL) {
+    // Generate the file
+    FileHelper::writeToFile($filePath, $output);
```

Comment:
I don't think you need a new helper for that. There is the SplFileObject if you want to use a OO method.
```
try {
    (new SplFileObject($filePath, "w"))->fwrite($output);
} catch (\RuntimeException $e) {
    // Most likely permission denied
    echo $e->getMessage();
}
```

Reply:
good to know